contracts/core/FollowNFT.sol

```diff
@@ -148,38 +122,39 @@ contract FollowNFT is LensNFTBase, IFollowNFT {
         returns (uint256)
     {
         if (blockNumber > block.number) revert Errors.BlockNumberInvalid();
-
         uint256 snapshotCount = _delSupplySnapshotCount;
+        if (snapshotCount == 0) return 0; // Returning zero since this means a delegation has never occurred
+        return _getSnapshotValueByBlockNumber(_delSupplySnapshots, blockNumber, snapshotCount);
+    }
 
-        if (snapshotCount == 0) {
-            return 0; // Returning zero since this means a delegation has never occurred
-        }
-
-        uint256 lower = 0;
-        uint256 upper = snapshotCount - 1;
-
-        // First check most recent delegated supply
-        if (_delSupplySnapshots[upper].blockNumber <= blockNumber) {
-            return _delSupplySnapshots[upper].value;
-        }
-
-        // Next check implicit zero balance
-        if (_delSupplySnapshots[lower].blockNumber > blockNumber) {
-            return 0;
-        }
-
-        while (upper > lower) {
-            uint256 center = upper - (upper - lower) / 2;
-            Snapshot memory snapshot = _delSupplySnapshots[center];
-            if (snapshot.blockNumber == blockNumber) {
-                return snapshot.value;
-            } else if (snapshot.blockNumber < blockNumber) {
-                lower = center;
-            } else {
-                upper = center - 1;
+    function _getSnapshotValueByBlockNumber(
+        mapping(uint256 => Snapshot) storage _shots,
```

Comment:
I would love to have a clearer name here, but `_snapshots` shadows the other field. If we find a way to rename `_snapshots` to `_snapshotsByBlahByBlah` and then `_shots`  to `_snapshots` would be great.

Reply:
We're keeping it as is, although this is a totally valid point, I think it's clear enough.